### PR TITLE
CXP-1044: Add acl on manage test apps

### PR DESCRIPTION
**Pull request by willy-ahva**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/Marketplace/TestApps/Controller/InternalApi/CreateTestAppsActionSpec.php [hunk 1]
```diff
@@ -97,14 +103,30 @@ public function it_redirects_to_the_root_if_the_request_does_not_come_from_ajax(
         $this->__invoke($request)->shouldBeLike(new RedirectResponse('/'));
     }
 
+    public function it_answer_an_access_denied_if_the_endpoint_is_not_granted_to_the_user(
```
Comment (masacc) [inline src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/Marketplace/TestApps/Controller/InternalApi/CreateTestAppsActionSpec.php:106]:
```suggestion
    public function it_answers_an_access_denied_if_the_endpoint_is_not_granted_to_the_user(
```

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/config/acl_groups.yml [hunk 2]
```diff
@@ -24,3 +24,6 @@ pim_enrich.acl_group.product:
     order: 110
 pim_enrich.acl_group.product_model:
     order: 120
+
+pim_system.acl_group.developer_mode:
+    order: 400
```
Comment (masacc) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/config/acl_groups.yml:29]: For my information, what are these lines of code doing?

**Review by masacc — APPROVED** [review]

**Comment by masacc** [comment]
Is it possible to have the ACL "manage test apps" without the ACL "manage apps" ? If not, should we check both of them each time ?

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/config/acl_groups.yml (hunk 2 above)
Comment (tseho) [reply src/Akeneo/Platform/Bundle/UIBundle/Resources/config/acl_groups.yml:29]: It automatically create a group of Acls for the User Role permissions form.

Comment (tseho) [reply src/Akeneo/Platform/Bundle/UIBundle/Resources/config/acl_groups.yml:29]: Friendly advice, the implementation of our ACLs is not a place in our code where it's nice to go.

**Comment by tseho** [comment]
> Is it possible to have the ACL "manage test apps" without the ACL "manage apps" ? If not, should we check both of them each time ?

Yes, it is.
Megane described each behavior in the ticket.

Screenshot_2022-01-13_15-28-01

You can create & connect a test without "manage apps".
We have to be sure this is possible.

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/config/acl_groups.yml (hunk 2 above)
Comment (willy-ahva) [reply src/Akeneo/Platform/Bundle/UIBundle/Resources/config/acl_groups.yml:29]: it adds a group to the acl meaning a new sub menu.

**Comment by tseho** [comment]
We don't want to have the developer mode enabled without the feature flag, at all.
The idea behind this is to forbid the developer mode or the test apps feature to be accessible in production, only on the demo cluster.

**Comment by willy-ahva** [comment]
> We don't want to have the developer mode enabled without the feature flag, at all. The idea behind this is to forbid the developer mode or the test apps feature to be accessible in production, only on the demo cluster.

done!

**Comment by willy-ahva** [comment]
> > Is it possible to have the ACL "manage test apps" without the ACL "manage apps" ? If not, should we check both of them each time ?
> 
> Yes, it is. Megane described each behavior in the ticket.
> 
> Screenshot_2022-01-13_15-28-01
> 
> You can create & connect a test without "manage apps". We have to be sure this is possible.

done!

File: src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/EventSubscriber/AclPrivilegesEventSubscriberSpec.php [hunk 3]
```diff
@@ -17,37 +17,80 @@
...
+        $developerModeFlag->isEnabled()->willReturn(false);
+        $event->getPrivileges()->willReturn($privileges);
+        $privileges->filter(Argument::any())->willReturn($filteredPrivileges);
+
+        $event->setPrivileges($filteredPrivileges)->shouldBeCalledOnce();
+
+        $this->disableAclIfFeatureIsDisabled($event);
+    }
+
+    public function it_filters_acls_if_both_flagq_are_disabled(
```
Comment (RiffVentura) [inline src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/EventSubscriber/AclPrivilegesEventSubscriberSpec.php:77]:
```suggestion
    public function it_filters_acls_if_both_flags_are_disabled(
```

**Review by RiffVentura — APPROVED** [review]

**Pull request merged**